src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -71,15 +105,62 @@ public void close() throws Exception {
     // Mostly for JUnit tests
     motor.close();
     motor2.close();
-    
   }
 
-  @Override
-  public void periodic() {
-    intakeMode.setString((isConeMode) ? "Cone Mode" : "Cube Mode");
-    // This method will be called once per scheduler run
+  public void changeFlightState() {
+    // Check if sensors are online
+    boolean coneSensorUp = timeOfFlightSensor.isSensor0Connected();
+    boolean cubeSensorUp = timeOfFlightSensor.isSensor1Connected();
+
+    // Log whether sensors are online
+    isConeSensor.log(coneSensorUp);
+    isCubeSensor.log(cubeSensorUp);
+
+    // Initialize both values to -1 and only change if sensors are up
+    int coneDistance = -1; 
+    int cubeDistance = -1;
+
+    // Check if cone sensor is up
+    if (coneSensorUp) {
+      coneDistance = timeOfFlightSensor.getDistance0();
+      hasCone.log(coneDistance <= mmConeActivationThreshold);
+    } else {
+      hasCone.log(false);
+    }
+
+    // Check if cube sensor is up
+    if (cubeSensorUp) {
+      cubeDistance = timeOfFlightSensor.getDistance1();
+      hasCube.log(false);
+    } else {
+      hasCube.log(false);
+    }
+
+    // Log distance values
+    coneDist.log(coneDistance);
+    cubeDist.log(cubeDistance);
+
+    // Change state (only if sensors are online)
+    if (coneDistance <= mmConeActivationThreshold && coneSensorUp) {
+      tofState = FlightStates.CONE;
+    } 
+    else if (cubeDistance <= mmCubeActivationThreshold && cubeSensorUp) {
+      tofState = FlightStates.CUBE;
+    } 
+    else if (coneSensorUp || cubeSensorUp) {
```

Comment:
This doesn't quite work. If only one sensor goes down it'll still return to IDLE instead of holding the last state.

Consider using another switch statement instead of if-else ladders to help determine the next state from the last one.